src/matter/interaction/InteractionServer.ts

```diff
@@ -224,30 +226,49 @@ export class InteractionServer implements ProtocolHandler<MatterDevice> {
         };
     }
 
-    handleSubscribeRequest(exchange: MessageExchange<MatterDevice>, { minIntervalFloorSeconds, maxIntervalCeilingSeconds, attributeRequests, keepSubscriptions }: SubscribeRequest): SubscribeResponse | undefined {
+    async handleSubscribeRequest(exchange: MessageExchange<MatterDevice>, { minIntervalFloorSeconds, maxIntervalCeilingSeconds, attributeRequests, eventRequests, keepSubscriptions }: SubscribeRequest, messenger: InteractionServerMessenger): Promise<SubscribeResponse | undefined> {
         logger.debug(`Received subscribe request from ${exchange.channel.getName()}`);
 
         if (!exchange.session.isSecure()) throw new Error("Subscriptions are only implemented on secure sessions");
         const session = exchange.session as SecureSession<MatterDevice>;
         const fabric = session.getFabric();
         if (fabric === undefined) throw new Error("Subscriptions are only implemented after a fabric has been assigned");
 
+        if ((!attributeRequests || attributeRequests.length === 0) && (!eventRequests || eventRequests.length === 0)) {
+            await messenger.sendStatus(StatusCode.InvalidAction);
```

Comment:
You can throw a typed error instead

Reply:
Ok we throw but will just work when the other PR is accepted, so now returns Failure, which is also okish